Orm/Xtensive.Orm/Orm/Rse/Providers/Compilable/OrderProviderBase.cs

```diff
@@ -66,11 +66,8 @@ protected override void Initialize()
       var comparisonRules = new ComparisonRules[Order.Count];
       for (int i = 0; i < Order.Count; i++) {
         var orderItem = Order[i];
-        var culture = CultureInfo.InvariantCulture;
         var column = Header.Columns[orderItem.Key];
-        var mappedColumn = column as MappedColumn;
-        if (mappedColumn != null && mappedColumn.ColumnInfoRef != null)
-          culture = mappedColumn.ColumnInfoRef.CultureInfo;
+        var culture = (column as MappedColumn)?.ColumnInfoRef.CultureInfo ?? CultureInfo.InvariantCulture;
```

Comment:
This might be a buggy line. You've changed the algorithm and ```null```s are not allowed to reach ```ComparisonRule```.  Some built-in comparison rules have ```CultureInfo``` set to ```null``` so ```null``` culture is valid value, I see it is taken to account all over the ```ComparisonRule``` class. ColumnInfoRef.CultureInfo can be null when ColumnInfo.CultureInfo is null which is theoretically possible but doesn't happen now, I'm not sure it should be but still. Did you see this?

If this causes problem it will be hard to find after few years.

Reply:
Yes, I thought about this semantic change.
Because `ColumnInfoRef` is converted to struct we cannot keep previous null-check.

But I tracked how `ColumnInfo.CultureInfo` may be set to `null`.  The only ways is setter property, which never called

Reply:
> The only way is property setter, which never called.

This is why I call it "hard to find". :) If somebody who doesn't know this thing or just not pays attention just set the property it will silently create a problem.

> Also we can change property type to nullable ```ColumnInfoRef? ColumnInfoRef.```
Is Checking whether ColumnInfoRef is default instance worse? Like so:
```
var culture = column is MappedColumn mColumn && mColumn.ColumnInfoRef != default
  ? mColumn.ColumnInfoRef.CultureInfo
  : CultureInfo.InvariantCulture;
```

I highly doubt that names of ColumnInfoRef that are both null will be valid case, right?

Reply:
`mColumn.ColumnInfoRef != default` is expension operation. It invokes `operator !=`, `ColumnInfoRef.Equals`.  

But we can check `ColumnInfoRef .TypeName != null`

Reply:
> var culture = column is MappedColumn mColumn && mColumn.ColumnInfoRef != default
>   ? mColumn.ColumnInfoRef.CultureInfo
>   : CultureInfo.InvariantCulture;

This was the first solution, but @AlexUstinov  argued against it https://github.com/DataObjects-NET/dataobjects-net/pull/207#discussion_r773553326